bridges/evm/contracts/example/NeoTCD.sol

```diff
@@ -0,0 +1,55 @@
+pragma solidity 0.5.14;
+pragma experimental ABIEncoderV2;
+
+interface Bridge {
+  struct VerifyOracleDataResult {
+    bytes data;
+    bytes32 codeHash;
+    bytes params;
+  }
+  function relayAndVerify(bytes calldata data) external returns (VerifyOracleDataResult memory result);
+}
+
+contract NeoTCD {
+  bytes32 public codeHash;
+  bytes public params;
+  uint256 public latestETHPrice;
+  uint256 public latestReportedBlock;
+
+  Bridge public bridge;
+
+  constructor(bytes32 _codeHash , bytes memory _params, address bridgeAddress) public {
+    codeHash = _codeHash;
+    params = _params;
+    bridge = Bridge(bridgeAddress);
+  }
+
+  function update(bytes memory _reportPrice) public {
+    Bridge.VerifyOracleDataResult memory result = bridge.relayAndVerify(_reportPrice);
+
+    require(result.codeHash == codeHash, "INVALID_CODEHASH");
+    require(keccak256(result.params) == keccak256(params), "INVALID_PARAMS");
+
+    uint256 _latestETHPrice;
```

Comment:
Can we re-use the library we wrote that decodes to `uint64[]`?

Reply:
done